pkg/vault/contracts/Router.sol

```diff
@@ -654,66 +654,83 @@ contract Router is IRouter, RouterCommon, ReentrancyGuardTransient {
             abi.decode(
                 _vault.unlock(
                     abi.encodeCall(
-                        Router.addLiquidityToBufferHook,
+                        Router.initializeBufferHook,
                         (
                             wrappedToken,
                             amountUnderlyingRaw,
                             amountWrappedRaw,
-                            msg.sender, // sharesOwner
-                            false // isBufferInitialized
+                            msg.sender // sharesOwner
                         )
                     )
                 ),
                 (uint256)
             );
     }
 
+    /**
+     * @notice Hook for initializing a vault buffer.
+     * @dev Can only be called by the Vault.
+     * @param wrappedToken Address of the wrapped token that implements IERC4626
+     * @param amountUnderlyingRaw Amount of underlying tokens that will be deposited into the buffer
+     * @param amountWrappedRaw Amount of wrapped tokens that will be deposited into the buffer
+     * @param sharesOwner Address that will own the deposited liquidity. Only this address will be able to
+     * remove liquidity from the buffer
+     * @return issuedShares the amount of tokens sharesOwner has in the buffer, expressed in underlying token amounts.
+     * (This is the BPT of an internal ERC4626 buffer)
+     */
+    function initializeBufferHook(
```

Comment:
Separating the initialize and add hooks is an improvement by itself (though it costs bytecode). It's clearer, more closely parallels the vault functions, and avoids passing flag arguments.